cns/service/main.go

```diff
@@ -187,6 +216,35 @@ func printVersion() {
 	fmt.Printf("Version %v\n", version)
 }
 
+// Try to register node with DNC when CNS is started in managed DNC mode
+func registerNode(httpRestService restserver.HTTPService, dncEP, infraVnet, nodeID string) {
+	var (
+		numCPU   = runtime.NumCPU()
+		response *http.Response
+		err      = fmt.Errorf("")
+		body     bytes.Buffer
+		httpc    = &http.Client{Timeout: time.Second * 30}
+	)
+
+	for err != nil {
+		response, err = httpc.Post(fmt.Sprintf("%s/%s/node/%s/cores/%d%s", dncEP, infraVnet, nodeID, numCPU, dncApiVersion), "application/json", &body)
+		if err == nil {
+			if response.StatusCode == http.StatusCreated {
+				var req cns.SetOrchestratorTypeRequest
+				json.NewDecoder(response.Body).Decode(&req)
+				httpRestService.SetNodeOrchestrator(&req)
+			} else {
+				logger.Errorf("[Azure CNS] Failed to register node with managed DNC with http status code: " + strconv.Itoa(response.StatusCode))
```

Comment:
Can you include node id and infravnetid in the log? It would help when looking at these logs in kusto